packages/go_router/lib/src/builder.dart

```diff
@@ -188,8 +192,12 @@ class RouteBuilder {
           routerNeglect, keyToPages, shellNavigatorKey, registry);
 
       // Build the Navigator
-      final Widget child = _buildNavigator(
-          onPopPage, keyToPages[shellNavigatorKey]!, shellNavigatorKey);
+      final Widget child = HeroControllerScope(
+        controller: route.heroController,
+        child: _buildNavigator(
+            onPopPage, keyToPages[shellNavigatorKey]!, shellNavigatorKey,
+            observers: observers),
+      );
```

Comment:
Why does only this `_buildNavigator` method call wrapped with `HeroControllerScope`? Shouldn't all `_buildNavigator` method calls be wrapped with `HeroControllerScope`?

Reply:
> 

The issue with the Hero Widgets not animating is only with the nested navigator and that is the ShellRoute, I've tested the top Navigator and for that the Hero widgets work fine, so `HeroControllerScope` is not required there.

Reply:
Option2 ;)

Reply:
For me, it causes a very significant performance hit. Not entirely sure what is the reason.